internal/querynode/task.go

```diff
@@ -600,3 +606,62 @@ func (r *releasePartitionsTask) Execute(ctx context.Context) error {
 func (r *releasePartitionsTask) PostExecute(ctx context.Context) error {
 	return nil
 }
+
+// releaseSegmentsTask
+func (r *releaseSegmentsTask) Timestamp() Timestamp {
+	if r.req.Base == nil {
+		log.Warn("nil base req in releaseSegmentsTask", zap.Any("collectionID", r.req.CollectionID))
+		return 0
+	}
+	return r.req.Base.Timestamp
+}
+
+func (r *releaseSegmentsTask) OnEnqueue() error {
+	if r.req == nil || r.req.Base == nil {
+		r.SetID(rand.Int63n(100000000000))
+	} else {
+		r.SetID(r.req.Base.MsgID)
+	}
+	return nil
+}
+
+func (r *releaseSegmentsTask) PreExecute(ctx context.Context) error {
+	return nil
+}
+
+func (r *releaseSegmentsTask) Execute(ctx context.Context) error {
```

Comment:
if streaming and historical don't have any data of the collection after releasing segment, why not release collection meta and clear query collection?

Reply:
fixed